pages/review/index.tsx

```diff
@@ -25,23 +31,31 @@ function ReviewPage({ hasUserAlreadyRegistered }: ReviewProps) {
       <Heading level="two">{t("review.title")}</Heading>
       <p>{t("review.description")}</p>
       <p>{t("internationalInstructions")}</p>
-      <ReviewFormContainer />
+      <ReviewFormContainer csrfToken={csrfToken} />
     </>
   );
 }
 
 export const getServerSideProps: GetServerSideProps = async ({
   query,
   req,
+  res,
 }) => {
   // We only want to get to this page from a form submission flow. If the page
   // is hit directly, then redirect to the home page.
   if (!query.newCard) {
     return homePageRedirect();
   }
   const hasUserAlreadyRegistered = !!req.cookies["nyplUserRegistered"];
+
+  const csrfToken = generateNewToken();
+  const newTokenCookieString = generateNewCookieTokenAndHash(csrfToken);
+  const tokenCookie = cookie.buildCookieHeader(newTokenCookieString);
+  res.setHeader("Set-Cookie", [tokenCookie]);
```

Comment:
Setting this header is the same as creating the cookie?

Reply:
Why do you need to do this on every page?

Reply:
it's not on every page, just the /account and /review pages. the token returned should be different for the different pages, so the cookie also needs to be updated.

Reply:
Gotcha. And then the newly generated token is compared on the request with the one set in the cookie. OK.